### PR TITLE
Fix PULL_REQUEST_TEMPLATE and improve test by removing unnecessary environment variable

### PR DESCRIPTION
**Pull request by thinkall**
## Why are these changes needed?

- Fix PULL_REQUEST_TEMPLATE
- Clean the environment variable `FLAML_MAX_CONCURRENT` set in the test in order to prevent potential errors.

## Checks

- [x] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR (note the same in integrated in our CI checks).
- [ ] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 ## Checks
 
-- I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR (note the same in integrated in our CI checks).
+- [ ] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR (note the same in integrated in our CI checks).
```
Comment (Borda) [inline .github/PULL_REQUEST_TEMPLATE.md:15]: This is not needed as it is part of CI

Comment (thinkall) [reply .github/PULL_REQUEST_TEMPLATE.md:15]: What about we hide this line?

Comment (Borda) [reply .github/PULL_REQUEST_TEMPLATE.md:15]: Sounds good to me

**Review by Borda — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 ## Checks
 
-- I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR (note the same in integrated in our CI checks).
+<!-- - [ ] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR (note the same in integrated in our CI checks). -->
```
Comment (sonichi) [inline .github/PULL_REQUEST_TEMPLATE.md:15]:
```suggestion
<!-- - I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR (note the same in integrated in our CI checks). -->
```

**Review by sonichi — APPROVED** [review]

**Pull request merged**